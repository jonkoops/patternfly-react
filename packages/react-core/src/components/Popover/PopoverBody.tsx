import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/Popover/popover';
import { css } from '@patternfly/react-styles';

export interface PopoverBodyProps extends HTMLProps<HTMLDivElement> {
  /** Popover body id */
  id: string;
  /** Popover body content */
  children: ReactNode;
  /** Classes to be applied to the popover body. */
  className?: string;
}

export const PopoverBody: FunctionComponent<PopoverBodyProps> = ({
  children,
  id,
  className,
  ...props
}: PopoverBodyProps) => (
  <div className={css(styles.popoverBody, className)} id={id} {...props}>
    {children}
  </div>
);
PopoverBody.displayName = 'PopoverBody';
