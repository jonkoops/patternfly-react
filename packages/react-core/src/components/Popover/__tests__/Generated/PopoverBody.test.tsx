/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { PopoverBody } from '../../PopoverBody';

it('PopoverBody should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<PopoverBody id={'string'} children={<div>ReactNode</div>} />);
  expect(asFragment()).toMatchSnapshot();
});
