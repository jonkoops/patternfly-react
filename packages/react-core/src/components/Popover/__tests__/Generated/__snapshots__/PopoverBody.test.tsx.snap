// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopoverBody should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-popover__body"
    id="string"
  >
    <div>
      ReactNode
    </div>
  </div>
</DocumentFragment>
`;
